cmd/k8s-operator/deploy/chart/templates/deployment.yaml

```diff
@@ -49,6 +49,7 @@ spec:
           image: {{ .Values.operatorConfig.image.repo }}{{- if .Values.operatorConfig.image.digest -}}{{ printf "@%s" .Values.operatorConfig.image.digest}}{{- else -}}{{ printf "%s" $operatorTag }}{{- end }}
           imagePullPolicy: {{ .Values.operatorConfig.image.pullPolicy }}
           env:
+            {{- if .Values.deploymentEnv }}{{ toYaml .Values.deploymentEnv | nindent 12 }}{{ end }}
```

Comment:
I would prefer us to discuss why folks might need it on a separate issue (and a separate PR)